src/main/java/testapp/adapter/memory/InMemoryStudentRepository.java

```diff
@@ -0,0 +1,38 @@
+package testapp.adapter.memory;
+
+import testapp.core.Student;
+import testapp.core.StudentAlreadyExistsException;
+import testapp.core.StudentRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class InMemoryStudentRepository implements StudentRepository {
+  private List<Student> studentList = new ArrayList<>();
+
+  @Override
+  public List<Student> findStudents() {
+    return studentList;
+  }
+
+  @Override
+  public void register(Student student) {
+    for (Student each : studentList) {
+      if (Objects.equals(each.getId(), student.getId())) {
```

Comment:
actually I meant that you should generate the Id here, and maybe check for taken username